### PR TITLE
Jshint fixes

### PR DESCRIPTION
**Pull request by hobinjk**
No more mutants^H^H^H^H^H^H^H warnings.

General workflow I followed is:
- `preferences.whatever.value == true` becomes `preferences.whatever.value`
- `something != false` and `something == true` -> `something`
- `something == false` and `something != true` -> `something`
- Variable scope/redeclarations get moved around as necessary
- In the event of a duplicate key the earlier declaration is deleted (as per js semantics), unless the earlier declaration is a preference separator
- Comma splices get fixed

The main thing to look for during review is places where the evaluation of `x != false` differs from `x` as described in https://dorey.github.io/JavaScript-Equality-Table/

### CODE REVIEW

File: Extensions/one_click_reply.js [hunk 1]
```diff
@@ -318,8 +319,11 @@ XKit.extensions.one_click_reply = new Object({
 			return s;
 		  }
 		  var  X = function(x, a, b) {return x.replace(new RegExp(a, 'g'), b)};
-		  var  R = function(a, b) {return s = X(s, a, b)};
-		  var blocks = '(table|thead|tfoot|caption|colgroup|tbody|tr|td|th|div|dl|dd|dt|ul|ol|li|pre|select'
+		var R = function(a, b) {
+			s = X(s, a, b);
+			return s;
+		};
```
Comment (bvtsang) [inline Extensions/one_click_reply.js]: Would the following work here?

```
var R = function(a, b) {
    return X(s, a, b);
};
```

File: Extensions/one_click_reply.js [hunk 2]
```diff
@@ -293,8 +293,9 @@ XKit.extensions.one_click_reply = new Object({
 				onload: function(response) {
 					// We are done!
 					XKit.interface.kitty.set(response.getResponseHeader("X-tumblr-kittens"));
+          var mdata = null;
```
Comment (bvtsang) [inline Extensions/one_click_reply.js]: Nitpick: indentation, as well as another below

File: Extensions/one_click_reply.js (hunk 1 above)
Comment (hobinjk) [inline Extensions/one_click_reply.js]: No, modifications to s are required because this code is really hacky

File: Extensions/postarchive.js [hunk 3]
```diff
@@ -383,9 +383,10 @@ XKit.extensions.postarchive = new Object({
 				    return;
 			    }
 
+					var m_obj = null;
```
Comment (bvtsang) [inline Extensions/postarchive.js]: Nitpick: indentation

File: Extensions/quick_tags.js [hunk 4]
```diff
@@ -226,10 +226,11 @@ XKit.extensions.quick_tags = new Object({
 
 		// Get the user tags.
 		var user_tags = XKit.storage.get("quick_tags","user_tags");
+    var user_tag_array = [];
```
Comment (bvtsang) [inline Extensions/quick_tags.js]: Nitpick: indentation

File: Extensions/reblog_yourself.js [hunk 5]
```diff
@@ -138,10 +138,10 @@ XKit.extensions.reblog_yourself = new Object({
 
 		if (!do_add) { return; }
 
+    var post_avatar = "";
```
Comment (bvtsang) [inline Extensions/reblog_yourself.js]: Nitpick: indentation

File: Extensions/stats.js [hunk 6]
```diff
@@ -350,29 +350,32 @@ XKit.extensions.stats = new Object({
...
 
-		for (var i=0;i<users.length;i++){
-			if (m_count == 4) {break; }
-			var perc = Math.round((users[i].count * 100) / posts.length);
-			var mx_html = 	"<a target=\"_BLANK\" href=\"http://" + users[i].url + ".tumblr.com/\"><div class=\"xkit-stats-blog\">" +
-						"<img src=\"https://api.tumblr.com/v2/blog/" + users[i].url + ".tumblr.com/avatar/32\" class=\"m_avatar\">" +
-						"<div class=\"m_title\">" + users[i].url + "</div>" +
+		users.forEach(function(user) {
+			if (m_count == 4) {
+				return;
```
Comment (bvtsang) [inline Extensions/stats.js]: I could be wrong, but I think `return` in a `forEach` doesn't function like a `break` in a `for` loop.

Compare the behaviors of the following:

```javascript
[1, 2, 3].forEach(function(i){
    if (i === 2) return;
    else console.log(i);
});
```

with

```javascript
for (var i = 1; i < 4; i++) {
    if (i === 2) break;
    else console.log(i);
}
```

File: Extensions/stats.js [hunk 7]
```diff
@@ -478,9 +481,9 @@ XKit.extensions.stats = new Object({
 
 		m_text = "<p><b>Top 4 blogs</b></p><ul>";
 
-		for (var i=0;i<4;i++){
-			var perc = Math.round((users[i].count * 100) / posts.length);
-			m_text = m_text + "<li><a href=\"" + users[i].url + ".tumblr.com\">" + users[i].url + "</a> <small>(" + perc + "%)</small></li>";
+		for (var user_i = 0; i < 4; i++){
```
Comment (bvtsang) [inline Extensions/stats.js]: `i` should be `user_i`

File: Extensions/view_on_dash.js [hunk 8]
```diff
@@ -299,26 +299,25 @@ XKit.extensions.view_on_dash = new Object({
...
+				rows.forEach(function(row_count) {
 					var shortest = 0;
-					var m_width = 500 / rows[i];
+					var m_width = 500 / row_count;
 
 					// Calculate the shortest!
 					var m_temp_photo = current_photo;
 
-					if (rows[i] >= 2) {
+					if (row >= 2) {
```
Comment (bvtsang) [inline Extensions/view_on_dash.js]: `row` should be `row_count`

File: Extensions/view_on_dash.js [hunk 9]
```diff
@@ -299,26 +299,25 @@ XKit.extensions.view_on_dash = new Object({
...
+				photo_post_inner_html = "<div class=\"photoset\" style=\"margin-bottom: 11px;\">";
 
 				var current_photo = 0;
 
-				for (var i=0;i<rows.length;i++) {
-
+				rows.forEach(function(row_count) {
 					var shortest = 0;
-					var m_width = 500 / rows[i];
+					var m_width = 500 / row_count;
```
Comment (bvtsang) [inline Extensions/view_on_dash.js]: `row_count` (here and parameter above) should be `row`

File: Extensions/view_on_dash.js (hunk 8 above)
Comment (bvtsang) [inline Extensions/view_on_dash.js]: `row` should be `row_count`

File: Extensions/xwidgets.js [hunk 10]
```diff
@@ -267,19 +267,19 @@ XKit.extensions.xwidgets = new Object({
...
+						m_caption = "Central Time";
+						break;
 					case -8:
-  						m_caption = "Pacific Time";
-  						break;
+						m_caption = "Pacific Time";
+						break;
 					default:
-  						m_caption = "GMT " + m_storage
+							m_caption = "GMT " + m_storage;
```
Comment (bvtsang) [inline Extensions/xwidgets.js]: Nitpick: indentation

File: Extensions/stats.js (hunk 6 above)
Comment (hobinjk) [inline Extensions/stats.js]: This (accidentally) ends up having the same behavior because m_count is never updated once it hits 4

**Comment by hobinjk** [comment]
@homu r=bvtsang

**Comment by hobinjk** [comment]
@homu r=bvtsang

**Pull request merged**